pgjdbc/src/main/java/org/postgresql/core/CachedQuery.java

```diff
@@ -55,7 +56,12 @@ public int getExecuteCount() {
 
   @Override
   public long getSize() {
```

Comment:
how frequently is this method called?
would it make more sense to calculate this value once (in constructor) and simply return that calculated value each time?

Reply:
I'm not an expert of the code base but are you sure it's twice in the life time ?
It seems to be twice per execution of the cached query (borrow / release) that's why it generates so many allocations.

Reply:
So that means 2x the number of times the "CachedQuery" is used over the life of the connection?

It was the "returning columns" bit that caused me some concern. Is this degree of exactness required? Would it be close enough to say `columnNames.length * 32` (i.e. assume 16 chars on average)?

Reply:
That's right. I stand corrected

Reply:
Technically speaking, `Arrays.equals` would have to be called, so iterating over columnNames one more time would not hurt much.

What do you think of just adding a field for size cache for `QueryWithReturningColumnsKey`?